src/Http/Response.php

```diff
@@ -2587,6 +2587,7 @@ protected function validateFile($path)
             throw new NotFoundException(__d('cake', 'The requested file contains `..` and will not be read.'));
         }
         if (!is_file($path)) {
+            deprecationWarning('Using non-absolute paths with Response::file() and withFile() is deprecated.');
```

Comment:
I'm a bit late, sorry... but wouldn't a `Folder::isAbsolute()` check be better suited? `is_file()` allows non absolute paths to slip through, given that the path would be looked up based on the current working directory. Also the warning would be triggered for all non existent files, irrespective of the type of the path.

Reply:
In 4.0 I was planning on deleting this block, so relative paths would continue working. The error message can be adjusted to reflect that the automatic scoping to APP is being removed.

Was your intention to also see relative file path support also removed?

Reply:
It was, yes, I should have elaborated a bit more on it.

Accepting relative paths assumes that the current working directory is a "safe" place, which IMHO isn't necessarily the case. Normally the CWD should be `webroot` folder, but that might vary.

What I personally would like to see is an exception being thrown for non absolute paths, so that people need to be explicit, and think about what they are trying to serve. Surely people would still be able to build "dangerous" paths that point to for example the `src` or `config` folder, but a mechanism that disallows certain paths would probably be a bit too much.

Reply:
Ok. I'll put up another pull request for that.